api/hasura/auth.ts

```diff
@@ -21,31 +21,33 @@ export default async function handler(req: VercelRequest, res: VercelResponse) {
       .replace('Bearer ', '')
       .split('|');
     const hashedToken = crypto.createHash('sha256').update(token).digest('hex');
-    const tokenRow = await adminClient.query({
-      personal_access_tokens: [
-        {
-          where: {
-            tokenable_type: { _eq: 'App\\Models\\Profile' },
-            id: { _eq: parseInt(expectedId) },
-            token: { _eq: hashedToken },
+    const tokenRow = await adminClient.query(
+      {
+        personal_access_tokens: [
+          {
+            where: {
+              tokenable_type: { _eq: 'App\\Models\\Profile' },
+              id: { _eq: parseInt(expectedId) },
+              token: { _eq: hashedToken },
+            },
           },
-        },
-        {
-          tokenable_id: true,
-        },
-      ],
-    });
-    const tokenableId = tokenRow.personal_access_tokens[0]?.tokenable_id;
+          {
+            tokenable_id: true,
+            profile: {
+              admin_view: true,
+              address: true,
+            },
+          },
+        ],
+        // cache query with ttl of 30s
+      },
+      { operationName: 'GetAuthToken @cached(ttl: 30)' }
```

Comment:
Nice . Caching in the auth layer is going to reduce alot of unnecessary overhead for every request , i wonder if we should push the time to the max 300s and force a cache refresh in a `delete_personal_access_tokens` event trigger